src/main/java/com/limechain/trie/TrieAccessor.java

```diff
@@ -0,0 +1,227 @@
+        }
+    }
+
+    @Override
+    public Optional<byte[]> find(Nibbles key) {
+        loadPathToKey(key);
+        return partialTrie.existingNode(key)
+                .map(NodeHandle::getUserData)
+                .map(NodeData::getValue);
+    }
+
+    @Override
+    public boolean delete(Nibbles key) {
+        loadPathToKey(key);
+        loadChildren(key);
+        markForRecalculation(key);
+        return partialTrie.removeValueAtKey(key);
+    }
+
+    @Override
+    public List<byte[]> findKeysByPrefix(Nibbles prefixSeek, int limit) {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public DeleteByPrefixResult deleteByPrefix(Nibbles prefix, Long limit) {
+        loadPathToKey(prefix);
+
+        NodeHandle<NodeData> parent = switch (partialTrie.node(prefix)) {
+            case Vacant<NodeData> vacant -> Optional.ofNullable(vacant.getClosestAncestorIndex())
+                    .map(partialTrie::nodeHandleAtIndex)
+                    .orElse(null);
+            case NodeHandle<NodeData> handle -> handle.getParent();
+        };
+
+        return Optional.ofNullable(parent)
+                .map(NodeHandle::getFullKey)
+                .map(parentKey -> prefixIndexInParent(parentKey, prefix))
+                .flatMap(parent::getChild)
+                .map(c -> lexicographicDelete(c, limit))
+                .orElse(new DeleteByPrefixResult(0, true));
+
+    }
+
+    private Nibble prefixIndexInParent(Nibbles parent, Nibbles prefix) {
+        for (int i = 0; i < prefix.size(); i++) {
+            if(prefix.get(i).equals(parent.get(i))) {
+                return parent.get(i);
+            }
+        }
+        return null;
+    }
+
+    private DeleteByPrefixResult lexicographicDelete(NodeHandle<NodeData> node, Long limit) {
+        loadChildren(node.getFullKey());
+        int deleted = 0;
+        for (Nibble nibble : Nibbles.ALL) {
+            if (limit != null && limit <= deleted) {
+                return new DeleteByPrefixResult(deleted, false);
+            }
+            NodeHandle<NodeData> child = node.getChild(nibble).orElse(null);
+            if (child != null) {
+                DeleteByPrefixResult childResult = lexicographicDelete(child, limit == null ? null : limit - deleted);
+                deleted += childResult.deleted();
+                if (!childResult.all()) {
+                    return new DeleteByPrefixResult(deleted, false);
+                }
+            }
+        }
+        if (partialTrie.clearNodeValue(node.getNodeIndex())) {
+            deleted++;
+        }
+        return new DeleteByPrefixResult(deleted, true);
+    }
+
+    public void persistAll() {
+        for (ChildTrieAccessor value : loadedChildTries.values()) value.persistAll();
+        loadedChildTries.clear();
+
+        if (partialTrie.isEmpty()) return;
+        lastRoot = getMerkleRoot(StateVersion.V1);
+        trieStorage.insertTrieStorage(partialTrie, StateVersion.V1);
```

Comment:
Those state versions will have to be refactored (won't be hardcoded to V1, but we'll have to figure out how to fetch and handle them). [Mental note, no changes required.]